proto/stakeibc/callbacks.proto

```diff
@@ -13,3 +13,10 @@ message DelegateCallback {
   uint64 depositRecordId = 2;
   repeated SplitDelegation splitDelegations = 3;
 }
+
+// ---------------------- Reinvest Callback ---------------------- //
+message ReinvestCallback {
```

Comment:
isn't it really a `bankSendCallback` that's used as part of the reinvestment cycle? Depending on how generalizable we'd like to make these protos, might make sense to rename

Reply:
I don't think we should generalize them - callbacks are currently designed to be at the level of abstraction of a single transaction "type", where the "type" of the transaction has logic that's specific to the application. Bank sends aren't specific to the application logic, so those callbacks could be handled in `OnAcknowledgementPacket`. The protos are just a convenient way to store function arguments for the different "types" of callbacks (e.g. you'd write a callback for a specific bank send that's relevant to some logic in a module, rather than a callback that's a catch-all for all bank sends)

Reply:
makes sense, ty for the writeup [](edited-with-Axolo)